### ISSUE TITLE
Bootstrap Backlog Process

### ISSUE THREAD

**Issue by rylev** [rust-lang/leadership-council#10]
This issue encompasses the following:
* [ ] Creating a location for tracking issues that require the attention of the Council. 
  * This will most likely be GitHub projects
* [ ]  Establish a simple process for folks outside the Council to bring items to the attention of the Council
* [ ] The previous item necessitates some sort of triaging process since it's likely some items might already be (at least partially) tracked or might be invalid for one reason or another.

Whatever process we decide on should have sign off by the rest of the Council as they will likely be interacting with the system on a somewhat regular basis.

**Comment by ehuss** [comment]
I'm not yet sure how or if GitHub projects might come into play for tracking things.

Here are some ideas for how to generally manage the council issue tracker:

# Issue tracking

The issue tracker should be used for tracking:

* Open Council and Committee tasks.
* Questions for the council.
* Proposals for new policy and process.

## Labeling

Labels used in the issue tracker:

* `question` — for questions to the council.
* `C-*` — labels for issues belonging to a specific Committee.
* `S-*` — labels for tracking the **status** of an issue:
    * `S-triage` — a council member needs to investigate the issue and either ask for more information or move it to some other status.
    * `S-needs-rfc` — this is a policy change that requires someone to develop an RFC.
    * `S-needs-info` — something that needs more information from the original requestor, or generally needs more research.
    * `S-needs-decision` — this is waiting on the Council to make a decision on the next steps.
    * `S-paused` — this is something that has been deferred to the future, and has been intentionally de-prioritized.
    * `S-blocked` — this is blocked, waiting on someone outside of the Council.
    * `S-active` — someone or some group is actively working on this.
    * `S-closed-inactive` — an issue closed because it has been inactive for a very long time and is unlikely to make progress.
* `A-*` — labels for different **areas** or topics:
    * `A-teams` — issues related to teams, such as forming new teams, team structure, etc.
    * `A-communication` — issues related to external communication, blog posts, etc.
    * `A-policy` — issues related to Project policy.
    * `A-process` — issues related to internal Council process.
    * `A-legal` — a legal-related issue.
    * `A-foundation` — related to the Rust Foundation.
* `P-*` — labels to denote the **priority**:
    * `P-urgent` — should be handled as soon as possible.
    * `P-high` — high priority.
    * `P-medium` — medium priority.
    * `P-low` — low priority.

## Templates

The following are some suggestions for adding GitHub issue templates.

(I think we could have additional templates over time, like maybe a "policy proposal", but I don't want to add too many right now.)

### Questions/Requests

* Auto-assign `S-triage` and `question` labels.
* Have a field for entering the question.

### Private feedback

A `contact_links` should be set up for private feedback, pointing to `council@rust-lang.org`.

### Security

GitHub automatically adds the org-wide security reporting option when at least one template option is offered (I believe).

## Triage

### Inbound triage

The entire Council is responsible for answering questions and moving issues out of the `S-triage` state. Ultimately the Librarians should ensure that issues are eventually triaged or to raise awareness of an issue to a Committee or the Council in order to find the next steps. This can be done informally, such as on the issue itself or in Zulip, or more formally such as being proposed for the sync meeting agenda (though that should only be done for rare cases where synchronous discussion and decision of the whole Council is needed).

Once `S-triage` is removed, if the issue isn't closed, one of the other status labels should be applied. If the issue is related to a specific Committee or area, those specific labels should be added. If more information is needed, the triager should ask for what is needed and set `S-needs-info`.

### Status triage

All Council Members are expected to participate in keeping issues up-to-date with their latest status. However, diffusion of responsibility can mean things don't get updated. The Librarians are responsible as a backstop for this:

* After approximately 3 months of inactivity, a Librarian should check that the issue accurately reflects its current status. If an issue needs to be redirected or delegated to someone to make it move forward, they should see if there is an obvious candidate.
* After approximately 1 year of inactivity, a Librarian should again check the issue accurately reflects its current status. If it seems like it is unlikely for the issue to get resolved in the foreseeable future, or seems to be indefinitely blocked on an outside party, they should post a comment to that effect (or ask for clarification) and notify that the issue will be closed during the next triage round if there is no further comment. A closed issue does not mean that it has been rejected, just that it is not active and is unlikely to be addressed. It can always be reopened if something changes. These issues will get the `S-closed-inactive` label to indicate that it was closed for this purpose.

## Backlog

Copy recommendations from https://rust-lang.github.io/rfcs/3392-leadership-council/initial-work-of-the-council.html to the issue tracker.

## TODO

I don't know what the process should be for nominating an issue for discussion or decision at a Council meeting. Should there be a "nominated" label that will automatically trigger its inclusion in the agenda? Or some other mechanism? Or is that even necessary?

**Comment by ehuss** [comment]
I'm going to close this as effectively done. Feel free to let me know if I misunderstood something or if you want to do anything differently.

* If anyone wants to create Projects to help track things or organize, feel free to do so at any time!
* Issue tracking is now documented at https://github.com/rust-lang/leadership-council/blob/main/procedures/issues.md, along with procedures for people to bring up issues or contact the Council, and the triaging of those requests.
* I have opened issues in this repo to start tracking known backlog issues and to providing a starting point for discussing those topics.
